plugins/GSdx/Renderers/DX11/GSRendererDX11.cpp

```diff
@@ -529,26 +529,47 @@ void GSRendererDX11::EmulateBlending()
 	uint8 blend_index  = uint8(((ALPHA.A * 3 + ALPHA.B) * 3 + ALPHA.C) * 3 + ALPHA.D);
 	int blend_flag = m_dev->GetBlendFlags(blend_index);
 
+	// SW free blend.
+	bool free_blend = !!(blend_flag & (BLEND_NO_BAR|BLEND_ACCU));
+
+	// Do the multiplication in shader for blending accumulation: Cs*As + Cd or Cs*Af + Cd
 	bool accumulation_blend = !!(blend_flag & BLEND_ACCU);
 
 	switch (m_sw_blending)
 	{
 		case ACC_BLEND_HIGH_D3D11:
 		case ACC_BLEND_MEDIUM_D3D11:
 		case ACC_BLEND_BASIC_D3D11:
-			sw_blending |= accumulation_blend;
+			sw_blending |= free_blend;
 			// fall through
 		default: break;
 	}
 
 	if (m_env.COLCLAMP.CLAMP == 0)
 	{
-		// fprintf(stderr, "%d: COLCLIP HDR mode%s\n", s_n, accumulation_blend ? " with accumulation blend" : "");
 		if (accumulation_blend)
+		{
+			// fprintf(stderr, "%d: COLCLIP HDR mode with accumulation blend\n", s_n);
 			sw_blending = true;
-		m_ps_sel.hdr = 1;
+			m_ps_sel.hdr = 1;
+		}
+		else if (sw_blending)
+		{
+			// So far only BLEND_NO_BAR should hit this path, it's faster than standard HDR algo
+			// fprintf(stderr, "%d: COLCLIP SW ENABLED (blending is %d/%d/%d/%d)\n", s_n, ALPHA.A, ALPHA.B, ALPHA.C, ALPHA.D);
+			m_ps_sel.colclip = 1;
+		}
+		else
+		{
+			// fprintf(stderr, "%d: COLCLIP HDR mode\n", s_n);
+			m_ps_sel.hdr = 1;
+		}
 	}
 
+#if 0
+	fprintf(stderr, "%d: BLEND_INFO: %d/%d/%d/%d. Clamp:%d. Prim:%d number %d (sw %d)\n",
+		s_n, ALPHA.A, ALPHA.B, ALPHA.C, ALPHA.D, m_env.COLCLAMP.CLAMP, m_vt.m_primclass, m_vertex.next, sw_blending);
+#endif
```

Comment:
Is this ifdef needed?

Reply:
Seems like this is for debugging, I personally don't mind if it'll be useful for some quick testing or anything. I've few of such stuff for PCRTC operations but under a different commented out macro in ``config.h``

Reply:
yeah but maybe we can just comment it?

Reply:
Sure, I don't mind either. The ``#ifdef macro`` case might be better in case he has a bunch of different debug statements to be enabled just with a single edit, whereas at such cases it might be a PITA to uncomment all other different statements. (but that doesn't seem to be the case here)